### PR TITLE
Slurm heterogeneous job support 7.8.x

### PR DESCRIPTION
**Pull request by hjoliver**
These changes partially address #3964 (will be closed by sibling PR to master)

See description, example and questions on the master version #3971 (example needs adapting to Cylc 7, of course)

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] (7.8.x branch) I have updated the documentation in this PR branch.
- [x] No dependency changes.

### CODE REVIEW

**Comment by hjoliver** [comment]
Job poll filtering added - see https://cylc.discourse.group/t/polling-slurm-hetjobs/292

**Comment by hjoliver** [comment]
(Job kill tested as working)

**Comment by RosalynHatcher** [comment]
The job polling commit works great.  Thank you very much for your help Hilary.

**Comment by hjoliver** [comment]
Unit tests stolen from master, with the new hetjob tests there, and adapted to Cylc 7 and Python 2.7.

**Comment by hjoliver** [comment]
@oliver-sanders and @dpmatthews  - this modifies the slurm handler so we'd better have a UK reviewer. Please review or reassign.

@dwsutherland - I've assigned you at this end because you actually use Slurm on our HPC, although I doubt that you seen heterogeneous jobs before (I hadn't). I think a quick sanity check will do. The code change is small (see description and example on the master version #3971); I've tested with Slurm here (requires "slurm_packjob" handler for our older Slurm version); and users at NCAS have confirmed that it works too.

**Needs to be reviewed by tomorrow so we can get this in the new release** (please :grin: )

File: doc/src/task-job-submission.rst [hunk 1]
```diff
@@ -278,6 +278,40 @@ directive explicitly if ``execution time limit`` is specified.
...
+   [runtime]
+       # run two heterogenous job components:
+       script = srun sleep 10 : sleep 30
+       [[my_task]]
+           [[[job]]]
+               batch system = slurm
+               execution time limit = PT1H
+           [[[directives]]]
+               --account = QXZ5W2
+               hetjob_0_--mem = 1G  # first prefix must be "0"
```
Comment (wxtim) [inline doc/src/task-job-submission.rst:295]: Possible user unclarity:
If I want more than one directive for `hetjob_0_` do I add a newline, or append to this line? - that is:
```
hetjob_0_--mem = 1G
hetjob_0_--gres = 6Apples
```
or 
```
hetjob_0_--mem = 1G --gres=5Peaches
```
Could you add a second directive for one or other hetjob to address this?

**Review by wxtim — APPROVED** [review]
I've made a single comment about the documentation, where I see potential for confusion.
As far as I can tell by eyeball it looks sane.
The tests look like they are testing what you want to test.

**Review by oliver-sanders — APPROVED** [review]
Took me a while to get my Cylc7 dev env working again but managed to get the tests passing.

File: doc/src/task-job-submission.rst (hunk 1 above)
Comment (hjoliver) [reply doc/src/task-job-submission.rst:295]: Good point @wxtim - it's the former. I'll update the documentation to add a second directive then merge this (as the change will not affect the code at all).

**Pull request merged**